_pages/documentation/general_docs/kconfig_build_system.md

```diff
@@ -0,0 +1,206 @@
+---
+layout: documentation
+title: Kconfig Build System
+doc: gem5 documentation
+parent: kconfig_build_system
+permalink: /documentation/general_docs/kconfig_build_system/
+---
+
+This guide is intended for advanced users who need to build gem5 with
+multiple ISAs or customize the build options, such as the Ruby memory protocol.
+Familiarity with the Kconfig system is required.
+
+## Build GEM5 with Kconfig Build System
+
+```bash
+scons [OPTIONS] Kconfig_command TARGET
+```
+
+Supported Kconfig commands include
+
+- defconfig
+- setconfig
+- menuconfig
+- guiconfig
+- listnewconfig
+- oldconfig
+- olddefconfig
+- savedefconfig
+
+The most common use of options are `defconfig`, `setconfig` and `menuconfig`.
+
+Use `scons --help` to list these commands, and for additional information.
+
+To build gem5 with Kconfig, there are now two steps.
+
+Initial configuration
+
+```bash
+scons defconfig gem5_build build_opts/ALL
+```
+
+This will create a configuration for the `gem5_build` build directory based on
+`build_opts/ALL`. The configuration is stored in the
+`gem5_build/gem5.build/config`.
+
+To build a target in a configured build directory, run a scons command similar
+to before.
+
+```bash
+scons -j$(nproc) gem5_build/gem5.opt
+```
+
+Note: In order to have backward compatibility for old build scheme. The users
+need to avoid use "build" directory for Kconfig build.
+
+To build gem5 Kconfig with customized Kconfig options, the additional step
+is required between **initial configuration** and **building the target**
+
+- Directly set Kconfig options
+
+```bash
+scons setconfig gem5_build USE_KVM=y
+```
+
+- set Kconfig options by `menuconfig`
+
+```bash
+scons menuconfig gem5_build
+```
+
+## Details of Kconfig Commands
+
+### defconfig
+
+Set up a config using values specified in a defconfig file, or if no value is
+given, use the default. The second argument specifies the defconfig file. All
+of default gem5 defconfig files are in `build_opts` directory. The user can
+also use their own defconfig files.
+
+```bash
+scons defconfig gem5_build build_opts/RISCV
+```
+
+Use the own defconfig file
+
+```bash
+scons defconfig gem5_build $HOME/foo/bar/myconfig
+```
+
+### setconfig
+
+Set values in an existing config directory as specified on the command line.
+
+The users or developers can get the Kconfig options via `menuconfig` or `guiconfig`
+
+For example, to enable gem5's built in systemc kernel:
+
+```bash
+scons setconfig gem5_build USE_SYSTEMC=y
+```
+
+### menuconfig
+
+Opens the menuconfig editor which will let you view and edit config values,
+and view help text. menuconfig runs in CLI.
+
+```bash
+scons menuconfig gem5_build
+```
+
+If is successful, the CLI will look like this
+
+![](/assets/img/kconfig/menuconfig.png)
+
+To view help text, for example, USE_ARM_ISA, will look like this
+
+![](/assets/img/kconfig/menuconfig_details.png)
+
+If the `gem5_build` directory is not exists. SCons will set up a build
```

Comment:
suggested grammar change:
If the `gem5_build` directory does not exist, SCons will set up a build directory at the path `gem5_build` with default options and then invoke menuconfig so you can set up its configuration.